packages/puppeteer-core/src/api/Page.ts

```diff
@@ -2563,14 +2565,7 @@ export abstract class Page extends EventEmitter<PageEvents> {
             ...scrollDimensions,
           });
           stack.defer(async () => {
-            if (viewport) {
-              await this.setViewport(viewport).catch(debugError);
-            } else {
-              await this.setViewport({
-                width: 0,
-                height: 0,
```

Comment:
I wonder if setting deviceScaleFactor here to 0 as well and reverting other changes would also fix the issue?

Reply:
I will try it.

Reply:
Set deviceScaleFactor to 0 does not fix the issue. I will later add a test case for the code change.